contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -0,0 +1,205 @@
+        }
+    }
+
+    /**
+     * @notice Updated bridge manager address
+     * @param _bridgeManager Bridge manager address
+     */
+    function setBridgeManager(
+        address _bridgeManager
+    ) external onlyBridgeManager {
+        if (_bridgeManager == address(0)) revert NotValidBridgeManager();
+        bridgeManager = _bridgeManager;
+        emit BridgeManagerUpdated(bridgeManager);
+    }
+
+    /**
+     * @notice Set the address of a wrapper using the token information if already exist
+     * @dev This function is used to allow any existing token to be mapped with
+     *      origin token.
+     * @notice If this function is called multiple times for the same existingTokenAddress,
+     * this will override the previous calls and only keep the last sovereignTokenAddress.
+     * @notice The tokenInfoToWrappedToken mapping  value is replaced by the new sovereign address but it's not the case for the wrappedTokenToTokenInfo map where the value is added, this way user will always be able to withdraw their tokens
+     * @param originNetwork Origin network
+     * @param originTokenAddress Origin token address, 0 address is reserved for ether
+     * @param sovereignTokenAddress Address of the sovereign wrapped token
+     * @param isNotMintable Flag to indicate if the wrapped token is not mintable
+     */
+    function setSovereignTokenAddress(
+        uint32 originNetwork,
+        address originTokenAddress,
+        address sovereignTokenAddress,
+        bool isNotMintable
+    ) external onlyBridgeManager {
+        // origin and sovereign token address are not 0
+        if (
+            originTokenAddress == address(0) ||
+            sovereignTokenAddress == address(0)
+        ) {
+            revert InvalidZeroAddress();
+        }
+        // originnetwork != current network, wrapped tokens are always from other networks
+        if (originNetwork == networkID) {
+            revert OriginNetworkInvalid();
+        }
+        // Compute token info hash
+        bytes32 tokenInfoHash = keccak256(
+            abi.encodePacked(originNetwork, originTokenAddress)
+        );
+        // Set the address of the wrapper
+        tokenInfoToWrappedToken[tokenInfoHash] = sovereignTokenAddress;
+        // Set the token info mapping
+        wrappedTokenToTokenInfo[sovereignTokenAddress] = TokenInformation(
+            originNetwork,
+            originTokenAddress
+        );
+        wrappedAddressIsNotMintable[sovereignTokenAddress] = isNotMintable;
+    }
+
+    /**
+     * @notice Remove the address of a remapped token from the mapping
+     * @notice It also removes the token from the isNotMintable mapping
+     * @param sovereignTokenAddress Address of the sovereign wrapped token
+     */
+    function removeSovereignTokenAddress(
+        address sovereignTokenAddress
+    ) external onlyBridgeManager {
+        delete wrappedTokenToTokenInfo[sovereignTokenAddress];
```

Comment:
maybe add check that if this is deleted there's a token with that tokenInfo!? ( so this was remapped already)!